basics/trey-hunner-zip-and-enumerate.md

```diff
@@ -76,9 +76,110 @@ c 3
 >>>
 ```
 
-Now you're ready to read [this awesome looping
-tutorial](http://treyhunner.com/2016/04/how-to-loop-with-indexes-in-python/).
-Read it now, then come back here and do the exercises.
+## zip
+
+What comes to your mind when you hear the word `zip`? A mechanism extensively used to tie two parts of something i.e. Shirt, Jacket etc. The `zip()` functions does pretty much same, it helps us tie two or more different items together.
+
+```python
+>>> users = ["Tushar", "Aman", "Anurag", "Sohit"]
+>>> uids = ["usr122", "usr123", "usr124", "usr125"]
+>>> user_details = zip(uids, users)
+>>> print(user_details)
+<zip object at 0x0000016938E311C0>
+>>>
+```
+
+After calling zip, an iterator is returned. In order to see the content wrapped inside, we need to first convert it to a list. Basically, `zip()` is an iterator, i.e. lazy: it gives the items as needed, instead of calculating them and storing them into memory all at once. So the zip object cannot show its elements before the elements are used, because it hasn't computed them yet.
```

Comment:
Maybe we should first show it with `print(list(user_details))`, and then do the same without `list()` and explain why that doesn't work as intended. This way the first successful print is as close to the explanation with shirt and jacket zippers as possible.